### DISCUSSION TITLE
Passing query parameter to exposed endpoint in custom view

### DISCUSSION THREAD

**Discussion by KNMohamed** [aminalaee/sqladmin#746] [Q&A] [answered]
In the documentation the following example is provided. 
```python
from sqladmin import BaseView, expose

class ReportView(BaseView):
    name = "Report Page"
    icon = "fa-solid fa-chart-line"

    @expose("/report", methods=["GET"])
    async def report_page(self, request):
        return await self.templates.TemplateResponse(request, "report.html")

admin.add_view(ReportView)
```

I'd like to know if it's possible for these **exposed** endpoints to support query parameters.
When I send a get request to "/reports?some_id=123" I get a 404 Not Found, which doesn't allow me to make use of any query parameters.

For example. I have a use case where I need a custom "report" view that has a dropdown of all our "reports". upon selection of a report, a table should populate with the values that correspond the select report.

I'm trying to make use of AJAX requests to dynamically update the data by passing the report_id to the exposed endpoint and populating the table.
Is there a way around this?

**Comment by aminalaee** [comment, marked as answer]
Shouldn't you pass request to "/admin/report" ? AFAIK the query parameter shouldn't interfere with the routing.

**Reply by KNMohamed** [reply]
You're right 😅, I was missing the /admin prefix, thanks for pointing that out.
